### PR TITLE
feat: Add batch export metrics to app_metrics

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

The view set that was not made for batch exports was (unsurprisingly) not returning batch export metrics. 

## Changes

Since re-usability has won the trade-off battle over complexity, here's a PR to return batch export "metrics" via the app metrics view set. The term "metrics" is in quotes, as event counts are kind of pointless in batch exports, but they fit the outdated app metrics model.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

Added a unit test.

### CODE REVIEW

File: posthog/api/app_metrics.py [hunk 1]
```diff
@@ -27,28 +32,24 @@ class AppMetricsViewSet(TeamAndOrgViewSetMixin, mixins.RetrieveModelMixin, views
...
-            uuid.UUID(kwargs["pk"])
+            rows = self.get_batch_export_runs_app_metrics_queryset(batch_export_id=kwargs["pk"])
+
+            dates = [row["dates"].strftime("%Y-%m-%d") for row in rows]
+            successes = [row["successes"] for row in rows]
+            failures = [row["failures"] for row in rows]
             return response.Response(
                 {
-                    "metrics": [
-                        {
```
Comment (tomasfarias) [inline posthog/api/app_metrics.py:35]: Not sure why this was a list, I made my thing match the schema returned by plugin metrics (which doesn't return a nested list). However, if we need a list, we can nest this again.

File: posthog/api/app_metrics.py [hunk 2]
```diff
@@ -27,28 +32,24 @@ class AppMetricsViewSet(TeamAndOrgViewSetMixin, mixins.RetrieveModelMixin, views
 
     def retrieve(self, request: request.Request, *args: Any, **kwargs: Any) -> response.Response:
         try:
-            # probe if we have a valid uuid, and thus are requesting metrics for a batch export
-            uuid.UUID(kwargs["pk"])
```
Comment (tomasfarias) [inline posthog/api/app_metrics.py:31]: I kept this check in the method I defined below.

File: posthog/api/app_metrics.py [hunk 3]
```diff
@@ -27,28 +32,24 @@ class AppMetricsViewSet(TeamAndOrgViewSetMixin, mixins.RetrieveModelMixin, views
...
+            rows = self.get_batch_export_runs_app_metrics_queryset(batch_export_id=kwargs["pk"])
+
+            dates = [row["dates"].strftime("%Y-%m-%d") for row in rows]
+            successes = [row["successes"] for row in rows]
+            failures = [row["failures"] for row in rows]
```
Comment (tomasfarias) [inline posthog/api/app_metrics.py:39]: QuerySets hit the database only "the first time we iterate over it". However, there are some minimal performance gains from doing this in a single loop, list comprehensions just were more natural for me to write. I can refactor.

File: posthog/api/app_metrics.py [hunk 4]
```diff
@@ -74,6 +75,50 @@ def error_details(self, request: request.Request, *args: Any, **kwargs: Any) ->
         error_details = AppMetricsErrorDetailsQuery(self.team, plugin_config.pk, filter).run()
         return response.Response({"result": error_details})
 
+    def get_batch_export_runs_app_metrics_queryset(self, batch_export_id: str):
+        """Use the Django ORM to fetch app metrics for batch export runs.
+
+        Attempts to (roughly) match the following (much more readable) query:
+        ```
+        select
```
Comment (tomasfarias) [inline posthog/api/app_metrics.py:83]: Took me 5 minutes to write the query and like a day to struggle with the ORM.

File: posthog/api/app_metrics.py [hunk 5]
```diff
@@ -74,6 +75,50 @@ def error_details(self, request: request.Request, *args: Any, **kwargs: Any) ->
...
+            date_trunc('day', last_updated_at) as dates,
```
Comment (tiina303) [inline posthog/api/app_metrics.py:84]: maybe add a comment about why we're using `last_updated_at` and what it means

**Review by tiina303 — APPROVED** [review]

File: posthog/api/app_metrics.py (hunk 5 above)
Comment (tomasfarias) [reply posthog/api/app_metrics.py:84]: Sure thing :+1:

File: posthog/api/app_metrics.py (hunk 4 above)
Comment (tiina303) [reply posthog/api/app_metrics.py:83]: do we have to use the ORM? we could just use a query too can't we?

**Pull request merged**